67_RayQueryGeometry/render.comp.hlsl

```diff
@@ -0,0 +1,77 @@
+#include "common.hlsl"
+
+struct SPushConstants
+{
+    uint64_t vertexBufferAddress;
+    uint64_t indexBufferAddress;
+};
+
+[[vk::push_constant]] SPushConstants pc;
+
+[[vk::binding(0, 0)]]
+RaytracingAccelerationStructure topLevelAS;
+
+[[vk::binding(0, 1)]]
+cbuffer CameraData
+{
+    SBasicViewParameters params;
+};
+
+[[vk::binding(0, 2)]] RWTexture2D<float4> outImage;
+
+[numthreads(WorkgroupSize, WorkgroupSize, 1)]
+void main(uint32_t3 threadID : SV_DispatchThreadID)
+{
+    uint2 coords = threadID.xy;
+    uint2 resolution;
+    outImage.GetDimensions(resolution.x, resolution.y);
+    uint2 texCoords = uint2(float(coords.x) / resolution.x, 1.0 - float(coords.y) / resolution.y);
+
+    if (any(coords >= resolution))
+        return;
+
+    float4 color = float4(0, 0, 0, 1);
+
+    const float4x4 invMVP = inverse(cameraData.params.MVP);
+    
+    float4 NDC = float4(texCoords * float2(2.0, -2.0) + float2(-1.0, 1.0), 0.0, 1.0);
+    float3 targetPos;
+    {
+        vec4 tmp = invMVP * NDC;
+        targetPos = tmp.xyz / tmp.w;
+        NDC.z = 1.0;
+    }
```

Comment:
compute all this on the CPU and send it as push constant to GPU

Reply:
ofc I know `texCoords` is a variable but the whole `NDC` can be precomputed as `scale*threadID.xy+offset`

Reply:
still a TODO